docs/rabbitmq.config.example

```diff
@@ -338,6 +372,26 @@
    %%
    %% {queue_index_embed_msgs_below, 4096},
 
+   %% The credits that a queue process is given by the message store.
```

Comment:
These paragraphs belong to a guide on rabbitmq.com. Comments to configuration keys can be brief descriptions with a link to more detailed docs.

Reply:
Here is [a good candidate](http://www.rabbitmq.com/persistence-conf.html). Might need some rearranging and editing but it tries to cover the same set of settings.

Reply:
I believe that contextual information is highly valuable when dealing with configuration properties. I agree that guides are essential for explaining how multiple configuration properties work together, but I strongly believe that we should keep essential details alongside configuration properties, in the config file context, without requiring users to jump out.

When I look at a canonical config example, I'm thinking:

1. What configs exist?
1. What do they mean?
1. Why would I want to change a config?

[The Cassandra config file example](https://github.com/apache/cassandra/blob/trunk/conf/cassandra.yaml) captures my sentiment well, [as does this less-known nginx config](https://github.com/gchef/nginx-cookbook/blob/master/attributes/default.rb).

I would really like to know what others think re the above.

Given the big picture, I will reduce the scope of this PR and just sync [PROJECT_ENV](https://github.com/rabbitmq/rabbitmq-server/blob/stable/Makefile#L10-L115) with `rabbitmq.config.example`. Single lines are in order, especially since I am yet to understand what every configuration property does.

Reply:
@gerhard Cassandra's config does not explain what Hinted Handoff is or why it exists in several paragraphs, it says "See http://wiki.apache.org/cassandra/HintedHandoff". A few of sentences per key makes sense but config examples should not end up being small guides. There are 2 config examples in master, would you duplicate all those "micro guides"? That's my point.

Reply:
There are number of guides already and most of config variables are documented in those guides, so we can keep doing that and add links here.